### PR TITLE
Update vector sample to reflect better practice

### PR DESCRIPTION
**Pull request by DuncanMcBain**
Some misleading comments were removed, and types names updated to match the 2020 specification.

### CODE REVIEW

File: samples/simple-example-of-vectors.cpp [hunk 1]
```diff
@@ -80,9 +72,9 @@ int main() {
        *   Vectors can also be scaled easily using operator overloads */
       cgh.parallel_for<vector_example>(range<3>(4, 4, 4), [=](item<3> item) {
         auto in = ptrA[item.get_linear_id()];
-        float w = in.w();
-        float3 swizzle = in.xyz();
-        float3 scaled = swizzle * w;
+        auto w = in.w();
+        vec<float,3> swizzle = in.xyz();
+        auto scaled = swizzle * w;
```
Comment (ProGTX) [inline samples/simple-example-of-vectors.cpp:77]: Why were these lines changed? It seems like they should have worked.

**Review by ProGTX — APPROVED** [review]

File: samples/simple-example-of-vectors.cpp (hunk 1 above)
Comment (DuncanMcBain) [reply samples/simple-example-of-vectors.cpp:77]: They worked fine, but I wanted to show off other ways of spelling the type, as well as `auto` working for per-element access.
